cloudpickle/cloudpickle.py

```diff
@@ -331,18 +367,22 @@ def save_function_tuple(self, func):
         save(_fill_function)  # skeleton function updater
         write(pickle.MARK)    # beginning of tuple that _fill_function expects
 
-        self._save_subimports(code, set(f_globals.values()) | set(closure))
+        self._save_subimports(
+            code,
+            itertools.chain(f_globals.values(), closure),
+        )
 
         # create a skeleton function object and memoize it
         save(_make_skel_func)
-        save((code, closure, base_globals))
+        save((code, compress_closure(func.__closure__), base_globals))
```

Comment:
Related to my comment above: I wanted to be able to tell the difference between None and `()` to preserve this across a cloudpickle round trip.